docs/website/docs/developers/general/github-actions.md

```diff
@@ -0,0 +1,315 @@
+  That script runs as part of the
+  [`setup.yml`](https://github.com/iree-org/iree/blob/main/.github/workflows/setup.yml)
+  action, which jobs can depend on like so:
+
+    ```yaml
+    jobs:
+      setup:
+        uses: ./.github/workflows/setup.yml
+
+      test:
+        needs: [setup]
+        if: contains(fromJson(needs.setup.outputs.enabled-jobs), 'test')
+        steps:
+          - ...
+    ```
+
+* Opt-in jobs can also be set up to run automatically if specific file paths
+  are modified, much like GitHub's `paths` and `paths-ignore` filters.
+
+To bypass the computed configuration of workflows to run, see the
+[CI behavior manipulation](./contributing.md#ci-behavior-manipulation) section
+of our contributing guide.
+
+### :octicons-server-16: Using GitHub-hosted and self-hosted runners
+
+We group runners into categories:
+
+1. GitHub-hosted runners: [standard (free)](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners)
+    * Workflow jobs should use these GitHub-hosted runners when at all possible.
+2. GitHub-hosted runners: [large (paid)](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners)
+    * We aren't currently using these, but they have been useful where
+      self-hosting is difficult.
+3. Self-hosted runners: CPU builders
+    * The core project maintainers sponsor a pool of powerful CPU build machines
+      used to build the core project and packages. To limit the load on this
+      pool, we restrict jobs using these runners to only the most essential.
+4. Self-hosted runners: special hardware
+    * Project contributors can offer their own hardware as self-hosted runners,
+      allowing tests to be run on that hardware at whatever frequency they
+      have the capacity to support.
+    * Due to how difficult it can be to keep self-hosted runners operating
+      reliably, and how access is limited to the group that maintains the
+      hardware, any jobs using these self-hosted runners must be optional and
+      easy to disable.
+    * Self-hosted runners can either be ephemeral (one job per runner,
+      compatible with
+      [autoscaling](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/autoscaling-with-self-hosted-runners)),
+      or persistent. Persistent runners can retain local build and artifact
+      caches to improve workflow time substantially.
+
+!!! info - "Contributing self-hosted runners"
+
+    Want to run tests on your own hardware as part of IREE's upstream CI? Get
+    in touch with us on one of our
+    [communication channels](../../index.md#communication-channels) and we'd be
+    happy to help discuss the options available.
+
+## :material-book-open-outline: Maintenance tips
+
+* Certain workflow failures are posted in the `#github-ci` channel in IREE's
+  Discord server
```

Comment:
```suggestion
  Discord server.
```
I think most listings are ended with a trailing punctuation so far.

Reply:
Yep, will change these all, thanks.

General guidelines I've been trying to follow: https://developers.google.com/style/lists#capitalization-and-end-punctuation (but I'm usually inconsistent)